packages/api/internal/handlers/template_delete.go

```diff
@@ -88,7 +89,39 @@ func (a *APIStore) DeleteTemplatesTemplateID(c *gin.Context, aliasOrTemplateID a
 		attribute.String("env.id", template.ID),
 	)
 
-	deleteJobErr := a.templateManager.DeleteInstance(ctx, template.ID)
+	// check if base env has snapshots
+	hasSnapshots, err := a.db.CheckBaseEnvHasSnapshots(ctx, template.ID)
+	if err != nil {
+		telemetry.ReportError(ctx, fmt.Errorf("error when checking if base env has snapshots: %w", err))
+		a.sendAPIStoreError(c, http.StatusInternalServerError, "Error when checking if base env has snapshots")
```

Comment:
this is user facing error, it should be something like
```suggestion
		a.sendAPIStoreError(c, http.StatusInternalServerError, "Error when checking if base template has paused sandboxes")
```